synapse/util/check_dependencies.py

```diff
@@ -128,6 +128,21 @@ def _incorrect_version(
         )
 
 
+def _no_reported_version(
+    requirement: Requirement, extra: Optional[str] = None
+):
+    if extra:
+        return (
+            f"Synapse {VERSION} needs {requirement} for {extra}, "
+            f"but can't determine {requirement.name}'s version."
```

Comment:
consistency nit: we don't include a `.` at the end of the other messages
```suggestion
            f"but can't determine {requirement.name}'s version"
```